packages/loot-core/src/server/sync/index.ts

```diff
@@ -503,8 +505,8 @@ export async function syncAndReceiveMessages(
   await receiveMessages(
     messages.map(msg => ({
       ...msg,
+      // TODO: This should be done inside the encoder (or at least it should return a different type)
       value: deserializeValue(msg.value as string),
-      timestamp: Timestamp.parse(msg.timestamp),
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏why are we removing this? Smells to me like this is not JUST a type change, but there are some functional changes too. Which is risky.

Reply:
We can remove this as https://github.com/actualbudget/actual/pull/1076/commits/9726e84a7fdb4009263069b91730e5abb36be79a expands the `Message` type to accept a `Timestamp` in its `timestamp` field.

The reason this was removed was that this wasn't satisfying the `Message` type which we had defined (it just wasn't erroring because `Timestamp.parse` returned `any`. We could then either expand the type or adjust the logic so when we pass this to `receiveMessages` then it's a valid `Message` and parse the timestamp later if necessary.

Reply:
I opted for the latter as we should be restricting our types where possible to avoid having to handle many different types, however we can restrict this type again in a later PR.